pyCRLD/Environments/HistoryEmbedding.py

```diff
@@ -346,10 +347,64 @@ def TransitionTensor(self):
     
     def RewardTensor(self):
         return histSjA_RewardTensor(self.baseenv, self.h)
-    
+
     def ObservationTensor(self):
```

Comment:
I've changed this function to be able to generate different observation tensors for each agent.